content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-flows.md

```diff
@@ -449,6 +449,178 @@ data class CreateBoardingTicketInitiator @JsonConstructor constructor(private va
 }
 ```
 
+## Write the `GiftVoucherToFriend` flow
+
+The `GiftVoucherToFriend` flow lets you to transfer the ownership of the issued voucher.
+
+Now that you've written the `CreateAndIssueMarsVoucher` and `CreateBoardingTicket` flows, try writing the `GiftVoucherToFriend` flow.
+
+You will need these variables:
+
+* voucherID
```

Comment:
@peterli-r3 is that correct? only one variable?

Reply:
GiftVoucherToFriend needs 2 parameters. One is voucherID, one is the new owner, I named it holder. 
https://github.com/corda/samples-java-corda5/blob/main/Tutorial/missionmars/workflows/src/main/java/net/corda/missionMars/flows/GiftVoucherToFriend.java#L84

Reply:
updated